### DISCUSSION TITLE
go: keep function signature after typing opening bracket

### DISCUSSION THREAD

**Discussion by gnur** [LunarVim/LunarVim#2495] [Q&A] [answered]
While typing a function to call I get this excellent preview of the function with its parameters and returned types:

<img width="871" alt="Screenshot 2022-04-21 at 10 29 20" src="https://user-images.githubusercontent.com/722329/164413684-c064a83c-11a9-4856-a913-37a97c515b69.png">

But as soon as I type the opening bracket it disappears: (I assume to make space for the autocomplete that follows if I type anything)
<img width="528" alt="Screenshot 2022-04-21 at 10 29 47" src="https://user-images.githubusercontent.com/722329/164413768-e6233415-1dd4-4a9e-bb75-517865075228.png">

Is there any way to keep the function signature next to the autocomplete?

**Comment by abzcoding** [comment, marked as answer]
atm i'm using lsp_signature plugin for this, 

![image](https://user-images.githubusercontent.com/10992695/164480589-51e0844a-11a7-4517-ae08-babf3c22cbb7.png)


but you can have it without as well

Using `ctrl-s` in insert mode for showing signature help
```lua
lvim.keys.insert_mode["<C-s>"] = "<cmd>lua vim.lsp.buf.signature_help()<cr>"

```

Also here is my config

https://github.com/abzcoding/lvim/blob/main/lua/user/lsp_signature.lua
